library/src/window_impl.cxx

```diff
@@ -44,6 +55,97 @@ class window_impl::internals
     return this->CachePath;
   }
 
+#if _WIN32
+  /**
+   * Helper function to detect if the
+   * Windows Build Number is equal or greater to a number
+   */
+  BOOL IsWindowsBuildNumberOrGreater(int buildNumber)
+  {
+    int windowsBuildNumber = 0;
+
+    // Receives the handle to Reg key
+    HKEY versionHKey;
+    LONG result = RegOpenKeyExA(HKEY_LOCAL_MACHINE,
+      "SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion", 0, KEY_READ, &versionHKey);
+
+    // Query value from the reg key
+    if (result == ERROR_SUCCESS)
+    {
+      // Receives code indicating the type of data stored in the reg value.
+      // CurrentBuildNumber is of type REG_SZ
+      DWORD type;
+      DWORD dataSize = MAX_VALUE_NAME;
+      CHAR data[MAX_VALUE_NAME]{};
+
+      RegGetValueA(versionHKey, "", "CurrentBuildNumber", RRF_RT_REG_SZ, &type, data, &dataSize);
+
+      if (result == ERROR_SUCCESS && type == REG_SZ)
+      {
+        std::from_chars(data, data + dataSize, windowsBuildNumber);
```

Comment:
need to check error code or are we sure it will parse to an `int`?

Reply:
I think there need to check error code